aea/cli/common.py

```diff
@@ -179,54 +184,69 @@ def try_to_load_agent_config(ctx: Context, is_exit_on_except: bool = True) -> No
             sys.exit(1)
 
 
-def _try_to_load_protocols(ctx: Context):
-    for protocol_public_id in ctx.agent_config.protocols:
-        protocol_name = protocol_public_id.name
-        protocol_author = protocol_public_id.author
-        logger.debug("Processing protocol {}".format(protocol_public_id))
-        protocol_dir = Path(
-            "vendor", protocol_public_id.author, "protocols", protocol_name
-        )
-        if not protocol_dir.exists():
-            protocol_dir = Path("protocols", protocol_name)
+def _load_env_file(env_file: str):
+    """
+    Load the content of the environment file into the process environment.
 
+    :param env_file: path to the env file.
+    :return: None.
+    """
+    load_dotenv(dotenv_path=Path(env_file), override=False)
+
+
+def _verify_or_create_private_keys(ctx: Context) -> None:
+    """
+    Verify or create private keys.
+
+    :param ctx: Context
+    """
+    path = Path(DEFAULT_AEA_CONFIG_FILE)
+    agent_loader = ConfigLoader("aea-config_schema.json", AgentConfig)
+    fp = path.open(mode="r", encoding="utf-8")
+    aea_conf = agent_loader.load(fp)
+
+    for identifier, _value in aea_conf.private_key_paths.read_all():
+        if identifier not in SUPPORTED_CRYPTOS:
+            ValueError("Unsupported identifier in private key paths.")
+
+    fetchai_private_key_path = aea_conf.private_key_paths.read(FETCHAI)
```

Comment:
everything below this could be refactored by having a map from ledger id to crypto classes.
Pseudocode

```python
_from_ledger_id_to_crytpo_class = {
    FETCHAI: FetchAICrypto,
    ETHEREUM: EthereumCrypto
}

for ledger_id, crypto_class in _from_ledger_id_to_crytpo_class.items()
    private_key_path = aea_conf.private_key_paths.read(ledger_id)
    if private_key_path is None:
        crypto_class().dump(file_pointer)
    else:
        # use crypto_class.load somehow
```

Of course the code can be different, this is just to give the idea.
We could also find a smarter way to define the mapping. Maybe built dynamically from the `*Crypto` classes in `aea.crypto` package.